crates/nu-cli/src/commands/str_/substring.rs

```diff
@@ -141,8 +140,20 @@ fn action(input: &Value, options: &Substring, tag: impl Into<Tag>) -> Result<Val
                 )
             })?;
 
-            let start: isize = options.0;
-            let end: isize = options.1;
+            let start: isize = if options.0 < 0 {
+                options.0 + len
+            } else {
+                options.0
+            };
+            let end: isize = if options.1 < 0 {
+                if options.1 < len {
```

Comment:
In this branch, `len` is positive and `options.1` is negative, so this condition will always be true. I also think `0` would be the better default when `options.1 < 0`. What do you think of

```rust
if options.1 >= -len {
    options.1 + len
} else {
    0
}
```

or a little more concise:

```rust
std::cmp::max(len + options.1, 0)
```

?

Reply:
Nice! I've updated the section with this.